File: javascript/q10.js
// Task: Fix the function to check if a string is a palindrome.

function isPalindrome(s) {
    let left = 0, right = s.length;
    while (left <= right) {
        if (s[left++] == s[right--]) return false;
    }
    return s;
}


/*
    Test case 1:
    Input: "racecar"
    Output: true

    Test case 2:
    Input: "hello"
    Output: false
*/

File: python/q10.py
# Task: Fix the function to check if a string is a palindrome.

def is_palindrome(s):
    left, right = 0, len(s)
    while left <= right:
        if s[left] == s[right]:
            return True
        left -= 1
        right += 1
    return False


"""
    Test case 1:
    input: 'racecar'
    output: True

    Test case 2:
    input: 'hello'
    output: False
"""
